esmvalcore/preprocessor/_time.py

```diff
@@ -172,8 +172,14 @@ def get_time_weights(cube):
         Array of time weights for averaging.
     """
     time = cube.coord('time')
-    time_thickness = time.bounds[..., 1] - time.bounds[..., 0]
-    time_weights = time_thickness * da.ones_like(cube.data)
+    coord_dims = cube.coord_dims('time')
+    time_weights = time.bounds[..., 1] - time.bounds[..., 0]
+    time_weights = time_weights.squeeze()
+    if coord_dims == ():
+        time_weights = np.broadcast_to(time_weights, cube.shape)
+    else:
+        time_weights = iris.util.broadcast_to_shape(time_weights, cube.shape,
+                                                    coord_dims)
```

Comment:
@schlunma cheers for finding and fixing le bug! Why not use the dask [equivalent](https://docs.dask.org/en/latest/array-api.html#dask.array.broadcast_to) and keep things lazy?

Reply:
Replaced `np.broadcast_to` by `da.broadcast_to`. However, for `iris.util.broadcast_to_shape` a `dask` equivalent does not exist.